php/tests/php_implementation_test.php

```diff
@@ -513,4 +513,128 @@ public function testPackedByteSize()
         TestUtil::setTestPackedMessage($m);
         $this->assertSame(166, $m->byteSize());
     }
+
+    public function testArrayConstructor()
+    {
+        $m = new TestMessage([
+            'optional_int32' => -42,
+            'optional_int64' => -43,
+            'optional_uint32' => 42,
+            'optional_uint64' => 43,
+            'optional_sint32' => -44,
+            'optional_sint64' => -45,
+            'optional_fixed32' => 46,
+            'optional_fixed64' => 47,
+            'optional_sfixed32' => -46,
+            'optional_sfixed64' => -47,
+            'optional_float' => 1.5,
+            'optional_double' => 1.6,
+            'optional_bool' => true,
+            'optional_string' => 'a',
+            'optional_bytes' => 'b',
+            'optional_enum' => TestEnum::ONE,
+            'optional_message' => [
```

Comment:
Can we just allow submessage? ```'optional_message' => new TestMessage_Sub([...])``` will be the same.

Reply:
It's not quite the same, as it requires the user to import the classes for all nested messages. Why not allow both? 

One of the main use cases we're trying to meet is when there are nested properties of messages, and importing / instantiating each class and object is unnecessarily tedious, e.g. 

```php
use Google\Cloud\Monitoring\V3\Point;
use Google\Cloud\Monitoring\V3\TypedValue;

$value = new TypedValue([
    'double_value' => new TypedValue(['value' => 123.45]),
]);
```

Could become

```php
use Google\Cloud\Monitoring\V3\Point;

$point = new Point(['double_value' => ['value' => 123.45]]);
```

Reply:
Allowing both makes the syntax confusing and causing unnecessary maintenance for us. The tediousness seems trivial to me.

Reply:
As requested I've removed this in b80f9f9. However, I do think the flexibility to use arrays for message properties would be useful in some cases, e.g. repeated fields and messages with a single property (as above).